staging/src/k8s.io/apimachinery/pkg/runtime/allocator.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package runtime
+
+import (
+	"sync"
+)
+
+// AllocatorPool simply stores Allocator objects to avoid additional memory allocations
+// by caching created but unused items for later reuse, relieving pressure on the garbage collector.
+//
+// Usage:
+//  memoryAllocator := serializer.AllocatorPool.Get().(*serializer.Allocator)
```

Comment:
Might be runtime.AllocatorPool.Get().(*serializer.Allocator)?

Reply:
it it actually 
```
//  memoryAllocator := runtime.AllocatorPool.Get().(*runtime.Allocator)
//  defer runtime.AllocatorPool.Put(memoryAllocator)
```
:)